index.js

```diff
@@ -27,16 +28,26 @@ module.exports = homebridge => {
     Service = homebridge.hap.Service
     Characteristic = homebridge.hap.Characteristic
     homebridge.registerAccessory('homebridge-mi-humidifier', 'MiHumidifier', MiHumidifier)
-}
+};
+
 
 class MiHumidifier {
-    constructor(log, config) {
+    constructor(log, config, api) {
         if (!config.ip) throw new Error('Your must provide IP address of the Humidifier');
         if (!config.token) throw new Error('Your must provide token of the Humidifier');
 
+        let options = { ...defaults, ...config };
+
+        this.humidifier = MiHumidifierFactory.create(log, config, api);
```

Comment:
`MiHumidifierFactory` will return null if the `config.model` is not supported by the Adapters. so there is no impact on the existing supported models.